### PR TITLE
Remove s2wasm

### PR DESCRIPTION
**Pull request by sbc100**
s2wasm is no longer used my emscripten and as far as I know now
as no other users.

### CODE REVIEW

**Comment by sbc100** [comment]
Probably worth holding off landing this for a week or two..

File: README.md [hunk 1]
```diff
@@ -64,7 +63,6 @@ This repository contains code that builds the following tools in `bin/`:
  * **wasm-opt**: Loads WebAssembly and runs Binaryen IR passes on it.
  * **asm2wasm**: An asm.js-to-WebAssembly compiler, using Emscripten's asm optimizer infrastructure. This is used by Emscripten in Binaryen mode when it uses Emscripten's fastcomp asm.js backend.
  * **wasm2asm**: A WebAssembly-to-asm.js compiler (still experimental).
- * **s2wasm**: A compiler from the `.s` format emitted by the new WebAssembly backend being developed in LLVM. This is used by Emscripten in Binaryen mode when it integrates with the new LLVM backend.
```
Comment (kripken) [inline README.md]: while we update this list, maybe worth adding `wasm-emscripten-finalize` and `wasm-link-metadata`?

File: src/wasm/wasm-emscripten.cpp [hunk 2]
```diff
@@ -159,6 +158,18 @@ static bool hasI64ResultOrParam(FunctionType* ft) {
   return false;
 }
 
+inline void exportFunction(Module& wasm, Name name, bool must_export) {
```
Comment (kripken) [inline src/wasm/wasm-emscripten.cpp]: why `inline`?

**Comment by sbc100** [comment]
Also removed all the .s files!

**Comment by BenHenning** [comment]
Is there a recommended alternative to s2wasm? I was using this for my own pipeline rather than using emscripten. Is there a new recommended build toolchian for folks who don't or can't depend on emscripten?

**Comment by jgravelle-google** [comment]
Upstream LLVM + LLD, which is what emscripten is using now.
Which is to say, `clang hello.c -o hello.wasm --target=wasm32 -nostdlib -Xlinker --no-entry -Xlinker --allow-undefined` should just work

**Comment by Markyparky56** [comment]
Don't mean to resurrect this issue or anything, but if you, like me, happen across this in a search to find a replacement to s2wasm then you'd better get familiar with wasm-ld. Check out `wasm-ld --help`. 

The `-Xlinker --no-entry` and `-Xlinker --allow-undefined` switches are just the beginning, wasm-ld includes switches for `--import-memory`, `--initial-memory=X` and even has the facility for explicit (or blanket) symbol exporting. By default it is set to `--no-export-dynamic` which means no functions will be exported, which seems a little odd but at least now you no longer have to deal with it exporting all your C++ name-mangled functions beside your nice clean extern "C" functions.

**Pull request merged**